### ISSUE TITLE
Semicolon in path prevents x360ce from recognizing "USB Joystick" (and potentially other) controllers

### ISSUE THREAD

**Issue by cerosrhino** [x360ce/x360ce#638]
The problem can be reproduced with a generic USB Joystick controller (I don't have any other controllers to test) and with either the 32-bit or 64-bit version of the software. I ran into it while trying to use x360ce with a Steam game that had a semicolon in its name. x360ce refused to work in this particular directory, but worked fine in others, and it took me a long time to find out why...

Steps to reproduce:

1. Create any directory with a semicolon (;) in its name or rename an existing directory to include this character.
2. Place the x360ce executable in said directory.
3. Run x360ce, allow it to create any necessary files.
4. The "General" tab for the controller is marked with red, the visual cues are grayed out. The controller seems to be completely unrecognized. My controller is a generic gamepad that is normally recognized as "USB Joystick", and in the tab labeled "USB Joystick" I can see that all the input from the controller is being received correctly, but the "General" tab is useless.

I have not verified whether it is just a front-end problem or if it actually prevents the software from working in-game (I tried to run the game and input wasn't working, but that might have been for other reasons, such as the game simply not supporting this way of input). However, the problem is easily reproducible, so I guess it's not that hard to find out.

Perhaps other special characters need to be taken into account as well?

**Comment by Squall-Leonhart** [comment]
semicolons are supposed to be blocked in file and folder names.....

**Comment by Nucleoprotein** [comment]
Nope, they are not, all character are allowed except /\:*"?<>| as well as NUL.

**Comment by Squall-Leonhart** [comment]
The semicolon is a reserved character that can be utilised in cmd/powershell functions.

Windows own api's ignore or truncate the character.

**Comment by Nucleoprotein** [comment]
Nope `mkdir "aaaa;"` works fine for me, at is should be, but it need to be escaped by quotes. It should work even in DOS COMMAND.COM.

https://en.wikipedia.org/wiki/Long_filename#Limits

https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#NAMING_CONVENTIONS

Semicolon was never reserved character.

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
